### PR TITLE
New database schema and setup

### PR DESCRIPTION
**Pull request by rahimftd**

### CODE REVIEW

File: db/config.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+dbConnection.once('open', function() {
+  console.log('Successfully connected to mongodb at mongodb://127.0.0.1:27017/penguin');
+});
+
+// Database schema
+var userSchema = mongoose.Schema({
+  username: String,
+  email: String,
+  funfact: String,
+  profileimage: String
```
Comment (nthaniel) [inline db/config.js]: camelCase here plz 
-love, Kevin

Comment (rahimftd) [inline db/config.js]: @nthaniel done!

File: db/db.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        } else {
+          resolve(newUser);
+        }
+      });
+    });
+  },
+
+  getAllUsers: function() {
+    return new Promise(function(resolve, reject) {
+      User.find(function(error, users) {
```
Comment (nthaniel) [inline db/db.js]: maybe: 
User.find().exec().then(function(users){//do stuff})
OR
User.find({}, function(error, users){//do stuff})

(unless your way already works in which case disregard)
http://mongoosejs.com/docs/api.html#model_Model.find

**Comment by rahimftd** [comment]
@nthaniel This is ready for you to check

File: db/db.js (hunk 2 above)
Comment (rahimftd) [inline db/db.js]: @nthaniel I used your second suggestion

File: db/db.js [hunk 3]
```diff
@@ -3,7 +3,7 @@
  * The mongodb server is configured in db/config.js
 */
 
-var User = require('./config').User;
+// var User = require('./config').User;
```
Comment (michelleheh) [inline db/db.js]: User is commented out but is required on line 21?

Comment (rahimftd) [inline db/db.js]: hmm... looks different in my code. let me close this one out and submit a new pull request.

Comment (rahimftd) [inline db/db.js]: @michelleheh actually I think you might be looking at an old commit. This pull request has a lot of commits, so some of the diffs might be outdated.

**Pull request merged**